#! /bin/sh -e

if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch) patch $pdir -f --no-backup-if-mismatch -p1 < $0;;
    -unpatch) patch $pdir -f --no-backup-if-mismatch -R -p1 < $0;;
    *)
	echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
	exit 1
esac
exit 0

# DP: Clarify documentation about substring expansion

--- bash/doc/bashref.texi~	2004-08-28 15:10:17.000000000 +0200
+++ bash/doc/bashref.texi	2004-08-28 18:27:16.000000000 +0200
@@ -1575,6 +1575,9 @@
 @var{length} must evaluate to a number greater than or equal to zero.
 If @var{offset} evaluates to a number less than zero, the value
 is used as an offset from the end of the value of @var{parameter}.
+Arithmetic expressions starting with a - must be separated by whitespace
+from the preceding : to be
+distinguished from the $@{@var{parameter}:@minus{}@var{word}@} expansion.
 If @var{parameter} is @samp{@@}, the result is @var{length} positional
 parameters beginning at @var{offset}.
 If @var{parameter} is an array name indexed by @samp{@@} or @samp{*},
